### PR TITLE
[QF-4205, QF-1730] Improve search results with Surah translations and Arabic verses

### PR DESCRIPTION
**Pull request by zonetecde**
## Summary

Created a shared component used by both **quick search** and **advanced search** to ensure consistent rendering in the search drawer and the full search page (`SearchResultText.tsx`).

Updated the rendering of search result items to follow these formats:

**Surah results**
```

1. Al Fatiha (The Opener)

```

**Verse results**
```

Arabic verse (Arabic surah name + Arabic verse key)
Translated verse (transliterated surah name + verse key)

```

Handled multiple edge cases, including:
- Mobile layout
- Arabic version of the site
- Arabic-only results depending on site language
- Other result types (e.g. `Page N`)

Added many Playwright regression tests.

Closes: [QF-4205](https://quranfoundation.atlassian.net/browse/QF-4205) and  [QF-1730](https://quranfoundation.atlassian.net/browse/QF-1730)

## Type of Change

- [X] ✨ New feature (non-breaking change)
- [X] ♻️ Refactoring (no functional changes)

## Test Plan

Added Playwright tests to ensure correct results are displayed in:
- The main search page
- The homepage search dropdown
- The search drawer

Verified that:
- Searching for a word always shows the Arabic verse **and** its translation, with the searched term highlighted
- Searching for a surah displays the transliterated name, its number, and the translation according to the site language

- [ ] Unit tests added/updated  
- [X] Integration tests added/updated  
- [X] Manual testing performed  

## Pre-Review Checklist

⚠️ This PR requires https://github.com/quran/search-service/pull/19 to work.

## Code Quality

- [X] Self-review completed (file by file)
- [X] Follows project style guidelines
- [X] No `any` types used (or justified if unavoidable)
- [X] No unused code, imports, or dead code
- [X] Inline comments explain complex logic (“why”)
- [X] Functions under 30 lines and single-responsibility

## Testing & Validation

- [X] All tests pass locally (`yarn test`)
- [X] Linting passes (`yarn lint`)
- [X] Build succeeds (`yarn build`)
- [X] Edge cases and error scenarios handled

## Documentation

- [X] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [X] Inline comments added for complex logic

## Localization (UI changes)

- [X] All user-facing text uses `next-translate`
- [ ] Only English locale files modified (Lokalise handles others)
- [X] RTL layout verified

## Accessibility (UI changes)

- [X] Semantic HTML elements used
- [X] ARIA attributes added where needed
- [X] Keyboard navigation works

## Screenshots / Videos

| Before | After |
| ------ | ----- |
|  |  |
| Arabic version<br/> | Arabic version<br/> |
| Arabic version (Arabic search)<br/> | Arabic version (Arabic search — no change)<br/> |
| Mobile<br/> | Mobile<br/> |

## Related PRs

Requires https://github.com/quran/search-service/pull/19

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [X] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4205]: https://quranfoundation.atlassian.net/browse/QF-4205?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[QF-1730]: https://quranfoundation.atlassian.net/browse/QF-1730?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Search/SearchResults/SearchResultItem/SearchResultText.tsx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+              dangerouslySetInnerHTML={{ __html: translationLine ?? '' }}
+            />
+          </div>
+        </div>
+      ) : (
+        <TextTag
+          className={classNames(textClasses.resultText, isArabic && textClasses.arabic)}
+          dir={singleLineDirection}
+          lang={singleLineLanguage}
+          dangerouslySetInnerHTML={{ __html: singleLineText ?? '' }}
```
[the remark this answers is not in the document]
Comment (zonetecde) [reply src/components/Search/SearchResults/SearchResultItem/SearchResultText.tsx:180]: @osamasayed should this be adressed ? we used `dangerouslySetInnerHTML` even before

[the remark this answers is not in the document]
Comment (osamasayed) [reply src/components/Search/SearchResults/SearchResultItem/SearchResultText.tsx:180]: No we should keep it as it is since the service is ours.

File: src/components/Search/SearchResults/SearchResultItem/SearchResultText.tsx [hunk 2]
```diff
@@ -0,0 +1,280 @@
+/* eslint-disable max-lines */
```
Comment (osamasayed) [inline src/components/Search/SearchResults/SearchResultItem/SearchResultText.tsx:1]: Please split it into more manigable parts, add functions and unit tests for these functions.

**Comment by osamasayed** [comment]
@zonetecde all good with this PR?

**Comment by zonetecde** [comment]
Yes in Shaa Allah
